File: app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseContract.java
package fr.free.nrw.commons.upload.license;

import fr.free.nrw.commons.BasePresenter;
import java.util.List;

/**
 * The contract with with MediaLicenseFragment and its presenter would talk to each other
 */
public interface MediaLicenseContract {

  interface View {

    void setLicenses(List<String> licenses);

    void setSelectedLicense(String license);

    void updateLicenseSummary(String selectedLicense, int numberOfItems);
  }

  interface UserActionListener extends BasePresenter<View> {

    void getLicenses();

    void selectLicense(String licenseName);
  }

}

File: app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseContract.kt
package fr.free.nrw.commons.upload.license

import fr.free.nrw.commons.BasePresenter

/**
 * The contract with with MediaLicenseFragment and its presenter would talk to each other
 */
interface MediaLicenseContract {
    interface View {
        fun setLicenses(licenses: List<String>?)

        fun setSelectedLicense(license: String?)

        fun updateLicenseSummary(selectedLicense: String?, numberOfItems: Int?)
    }

    interface UserActionListener : BasePresenter<View> {
        fun getLicenses()

        fun selectLicense(licenseName: String)

        fun isWLMSupportedForThisPlace(): Boolean
    }
}
